### PR TITLE
add new endpoint - list repository events

### PR DESCRIPTION
**Pull request by josevi96**
New endpoint to list the events of a particular repository with `listPublicRepositoryEvents` using the owner and the repo's name.
Github docs:
https://docs.github.com/en/rest/activity/events#list-repository-events

### CODE REVIEW

**Comment by josevi96** [comment]
Hi folks!, I was thinking about merging the PublicRepositoryEvent and the PublicOrganizationEvent as they are almost the same but as they are different entities in the github API I decided to separate them, what do you think?

File: github4s/shared/src/main/scala/github4s/Decoders.scala [hunk 1]
```diff
@@ -265,6 +265,25 @@ object Decoders {
       )
     )
 
+  implicit val decodePublicRepositoryEvent: Decoder[PublicRepositoryEvent] =
```
Comment (fedefernandez) [inline github4s/shared/src/main/scala/github4s/Decoders.scala:268]: `PublicRepositoryEvent` model looks similar to `PublicOrganizationEvent`. Can you please check the API and see if we can reuse the model?

Comment (josevi96) [reply github4s/shared/src/main/scala/github4s/Decoders.scala:268]: Yes they share the same schema, we can merge both models into the same one

**Review by necosta — APPROVED** [review]
Looks good. Great contribution 💪

**Review by fedefernandez — APPROVED** [review]

**Pull request merged**